File: Java/civitas/src/civitas/SorpresaIraCasilla.java
package civitas;
import GUI.Dado;

import java.util.ArrayList;

public class SorpresaIraCasilla extends Sorpresa{
    private Tablero tablero;
    private int valor;
    
    public SorpresaIraCasilla(Tablero _tablero, int _valor, String _texto){
        super(_texto);
        tablero = _tablero;
        valor = _valor;
    }
    
    @Override
     public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual, todos) == true){
            super.informe(actual, todos);
            Casilla casillaActual = tablero.getCasilla(todos.get(actual).getNumCasillaActual());
            int numDado = Dado.getInstance().tirar();
            int tirada = tablero.calcularTirada(todos.get(actual).getNumCasillaActual(), numDado);
            int casillaFinal = tablero.nuevaPosicion(todos.get(actual).getNumCasillaActual(), tirada);
            todos.get(actual).moverACasilla(casillaFinal);
            casillaActual.recibeJugador(actual, todos);
        }
    }

    @Override
    public String toString(){
        return "Sorpresa{ " + super.getTexto() + ", valor " + Integer.toString(valor) + " }";
                 
    }
}

File: Ruby/civitas/pkg/civitas-0.0.1/lib/sorpresa_ir_a_casilla.rb
#encoding: utf­8
require_relative "sorpresa.rb"
module Civitas
class SorpresaIrACasilla < Sorpresa
  def initialize(_tablero, _valor, _texto)
    super(_texto)
    @tablero = _tablero
    @valor = _valor
  end
  
  def aplicar_a_jugador(actual,todos)
      if(jugador_correcto(actual, todos))
        informe(actual, todos)
        casilla = @tablero.get_casilla(todos.at(actual).num_casilla_actual)
        num_dado = Dado.new.tirar()
        tirada = @tablero.calcular_tirada(todos.at(actual).num_casilla_actual, num_dado)
        casilla_final = @tablero.nueva_posicion(todos.at(actual).num_casilla_actual, tirada)
        todos.at(actual).mover_a_casilla(casilla_final)
        casilla.recibe_jugador(actual, todos)
      end
    end
    
 def to_ss()
    return @texto
  end
end
end
